app/helpers/authentication_helper.rb

```diff
@@ -1,10 +1,10 @@
 module AuthenticationHelper
 
   def logged_in?
-    !!session[:user_id]
+    render status: :unauthorized unless session[:user_id].present?
   end
 
   def current_user
-    @current_user ||= User.find_by(id: session[:user_id]) if !!session[:user_id]
+    @current_user = User.find_by(id: session[:user_id])
```

Comment:
Please use memoization so you don't make a sql query every time you call `current_user`